### PR TITLE
Fixed ignore in smoothed cross entropy metric and added a test. Thank…

### PR DESCRIPTION
**Pull request by fhieber**
Fixes the use of ignore in `cross_entropy_smoothed` as brought up by #200 .
Thanks @kobenaxie !

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./pre-commit.sh` or manual run of pylint & mypy)
- [x] You have considered writing a test
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [x] Updated CHANGELOG.md

### CODE REVIEW

File: sockeye/loss.py [hunk 1]
```diff
@@ -144,20 +144,20 @@ def __init__(self,
...
-    def _cross_entropy_smoothed(self, pred, label, ignore):
+    def cross_entropy_smoothed(self, pred, label, ignore):
         label_dist = mx.nd.one_hot(indices=label.astype(dtype='int32'),
                                    depth=self.loss_config.vocab_size,
                                    on_value=1.0 - self.loss_config.label_smoothing,
                                    off_value=self.loss_config.label_smoothing /
                                              (self.loss_config.vocab_size - 1.0))
-        label_dist = mx.nd.where(ignore, label_dist, mx.nd.zeros_like(label_dist))
-        loss = label_dist * (- mx.nd.log(pred + 1e-8)) # pylint: disable=invalid-unary-operand-type
+        label_dist = mx.nd.where(1 - ignore, label_dist, mx.nd.zeros_like(label_dist))
```
Comment (mjdenkowski) [inline sockeye/loss.py]: Good catch!
